cluster/client.go

```diff
@@ -519,31 +524,34 @@ func (c *Client) retry(command *proto.Command, nodeAddr string, timeout time.Dur
 	return p, nil
 }
 
-func writeCommand(conn net.Conn, c *proto.Command, timeout time.Duration) error {
+func writeCommand(conn net.Conn, c *proto.Command, timeout time.Duration) (err error, retriable bool) {
 	p, err := pb.Marshal(c)
 	if err != nil {
-		return fmt.Errorf("command marshal: %w", err)
+		return fmt.Errorf("command marshal: %w", err), false
 	}
 
 	// Write length of Protobuf
 	if err := conn.SetDeadline(time.Now().Add(timeout)); err != nil {
-		return err
+		// This error is retriable, as we haven't written the command yet
+		return err, true
 	}
 	b := make([]byte, protoBufferLengthSize)
 	binary.LittleEndian.PutUint64(b[0:], uint64(len(p)))
 	_, err = conn.Write(b)
 	if err != nil {
-		return fmt.Errorf("write length: %w", err)
+		// This error is retriable, as we haven't written the command yet
+		return fmt.Errorf("write length: %w", err), true
 	}
-	// Write actual protobuf.
+	// Write actual protobuf
 	if err := conn.SetDeadline(time.Now().Add(timeout)); err != nil {
-		return err
+		return err, true
 	}
 	_, err = conn.Write(p)
 	if err != nil {
-		return fmt.Errorf("write protobuf bytes: %w", err)
+		// This error is not retriable, as we might have sent the entire command
+		return fmt.Errorf("write protobuf bytes: %w", err), false
 	}
-	return nil
+	return nil, false
```

Comment:
The issue I have with this is that we don't know if it's retryable. Perhaps the network completely failed, not a single packet made it to the other side, and we could retry the write.

Before we make this change, I think it would be better to instrument the code better to count if writeCommand is ever failing (and in which place). If we detect no errors during your test run, and yet we still see results we can't account for, then this change won't help.

I do think this function is a bit naive however, for the reasons you are touching on here.